### DISCUSSION TITLE
Can not get Plugin [id: 'com.varabyte.kobweb.application', version: '0.18.2']

### DISCUSSION THREAD

**Discussion by liweikui** [varabyte/kobweb#577] [Q&A]
When I follow the Readme file step by step in Windonds 10, I get the error as following when I run the command: kobweb run.
someone can tell me why? thanks!

Plugin [id: 'com.varabyte.kobweb.application', version: '0.18.2'] was not found in any of the following sources:

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.api.plugins.UnknownPluginException: Plugin [id: 'com.varabyte.kobweb.application', version: '0.18.2'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (No included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'com.varabyte.kobweb.application:com.varabyte.kobweb.application.gradle.plugin:0.18.2')
  Searched in the following repositories:
    Gradle Central Plugin Repository
    maven(https://maven.pkg.jetbrains.space/public/p/compose/dev)
    maven2(https://us-central1-maven.pkg.dev/varabyte-repos/public)
	at org.gradle.plugin.use.resolve.internal.PluginResolutionResult.getFound(PluginResolutionResult.java:112)
	at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.resolvePluginRequest(DefaultPluginRequestApplicator.java:192)
	at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugins(DefaultPluginRequestApplicator.java:97)
	at org.gradle.kotlin.dsl.provider.PluginRequestsHandler.handle(PluginRequestsHandler.kt:45)
	at org.gradle.kotlin.dsl.provider.StandardKotlinScriptEvaluator$InterpreterHost.applyPluginsTo(KotlinScriptEvaluator.kt:212)
	at org.gradle.kotlin.dsl.execution.Interpreter$ProgramHost.applyPluginsTo(Interpreter.kt:385)

**Comment by liweikui** [comment]
the config in settings.gradle.kts : 
pluginManagement {
    repositories {
        gradlePluginPortal()
        maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
        maven("https://us-central1-maven.pkg.dev/varabyte-repos/public")
    }
}

the config in build.gradle.kts
plugins {
    alias(libs.plugins.kotlin.multiplatform) apply false
}

subprojects {
    repositories {
        mavenCentral()
        maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
        google()
        maven("https://us-central1-maven.pkg.dev/varabyte-repos/public")
    }
}

I can open https://maven.pkg.jetbrains.space/public/p/compose/dev in browser, but I can not open https://us-central1-maven.pkg.dev/varabyte-repos/public in browser.
I have checked my network, I do not have any proxy configuration.

**Comment by bitspittle** [comment]
The URL to the entry registration for the artifact you are trying to pull down is: 

https://console.cloud.google.com/artifacts/maven/varabyte-repos/us-central1/public/com.varabyte.kobweb.gradle:application/0.18.2?project=varabyte-repos

Are you on a home network or a corporate network?

**Comment by liweikui** [comment]
> The URL to the entry registration for the artifact you are trying to pull down is:
> 
> https://console.cloud.google.com/artifacts/maven/varabyte-repos/us-central1/public/com.varabyte.kobweb.gradle:application/0.18.2?project=varabyte-repos
> 
> Are you on a home network or a corporate network?

I am on a corporate network. I can open the URL you mentioned in browser, How can I get it in my kobweb project? Thanks!

**Reply by bitspittle** [reply]
Sadly, there's a decent chance this is something that is being blocked by your corporate network.

If you want a workaround, you can build Kobweb yourself locally:

```bash
$ git clone https://github.com/varabyte/kobweb
$ cd kobweb
$ ./gradlew publishMavenLocal
```

Then, in your own project, both in settings.gradle.kts and your relevant build.gradle.kts files, add

```kotlin
repositories {
    mavenLocal {
        content {
            includeGroup("com.varabyte.kobweb")
        }
    }

```

When Kobweb hits 1.0, we'll start publishing artifacts to maven central and the Gradle plugin repo, so this should not be a problem forever. However, these providers add a lot of friction to the release step, so we're putting it off while still expecting to do regular, active development that sometimes needs to move very quickly.

**Reply by liweikui** [reply]
Thank you very much! I got my Kobweb project ran by these steps. But differently, after I built Kobweb locally, I just add mavenLocal() in repositories, not the whole content above.

**Reply by bitspittle** [reply]
Keep in mind this solution is not recommended as a sustainable, scalable way forward. It can be good if you're trying to prototype something or work on something on your own, but if you plan to ask a team to use Kobweb, they will not all be happy syncing and compiling the framework themselves. 

In your corporate setting, I think you have three choices: 
1) ask your IT department to allowlist our maven URL
2) create your own maven repository internally inside your company that you maintain and upload Kobweb artifacts to, and have people use that url
3) Don't use Kobweb yet, it might be more friction than it is worth for now :(

**Comment by liweikui** [comment]
> > The URL to the entry registration for the artifact you are trying to pull down is:
> > https://console.cloud.google.com/artifacts/maven/varabyte-repos/us-central1/public/com.varabyte.kobweb.gradle:application/0.18.2?project=varabyte-repos
> > Are you on a home network or a corporate network?
> 
> I am on a corporate network. I can open the URL you mentioned in browser, How can I get it in my kobweb project? Thanks!

When I try to open the url, https://us-central1-maven.pkg.dev/varabyte-repos/public, in browser, I always get a 404 error.

**Reply by bitspittle** [reply]
Sure, you are probably not expected to visit that URL directly.
